frontend/src/components/UserProfile/UserProfile.tsx

```diff
@@ -64,19 +64,14 @@ const _UserProfile = ({
 
   const handleFollowClick = (e: any) => {
     setLoading(true);
-    if (isFollowing) {
-      unfollowUser(user?.id, cookies.token, () => {
-        setLoading(false);
-        setIsFollowing(false);
-        fetchUserWithFollows(cookies.token, id);
-      });
-    } else {
-      followUser(user?.id, cookies.token, () => {
-        setLoading(false);
-        setIsFollowing(true);
-        fetchUserWithFollows(cookies.token, id);
-      });
-    }
+    let callback = () => {
+      setLoading(false);
+      setIsFollowing(false);
+      fetchUserWithFollows(cookies.token, id);
+    };
+    isFollowing
+      ? unfollowUser(user?.id, cookies.token, callback)
+      : followUser(user?.id, cookies.token, callback);
```

Comment:
This is good for working, there is another way to simplify this. This topic is related to [destructuring in js](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
```js
let data = {
   user_id: user?.id,
   cookies_token: cookies.token,
   callback: () => {
      setLoading(false);
      setIsFollowing(false);
      fetchUserWithFollows(cookies.token, id);
   }
} 

isFollowing ? unfollowUser(data) : followUser(data);
```
And after this, update your `followUser` and `unfollowUser` method that accepts on parameter.
```js
followUser({ user_id, cookies_token, callback}){
//code
}
unfollowUser({ user_id, cookies_token, callback}){
//code
}
```